File: Algorithms/Sorting Algorithms/Bubble Sort/bubble_sort.java
//"n" is the size of the structure to be sorted by the method
//"swap" performs the exchange of position between two elements 

/**
 * Best Case:
 *      Comparisons = O(n)
 *      Swaps = O(1)
 * 
 * Average Case:
 *      Comparisons = O(n²)
 *      Swaps = O(n²)
 * 
 * Worst Case:
 *      Comparisons = O(n²)
 *      Swaps = O(n²)
 */

public class bubble_sort{
    public void bubble_sort(){
        for(int i= (n-1); i > 0; i--){
            for(int j= 0; j < i; j++){
                if(array[j] > array[j + 1]){
                    swap(j, j+1);
                }
            }
        }
    }
}

File: Algorithms/Sorting Algorithms/Bubble Sort/bubble_sort.py
# Best Case:
#      Comparisons = O(n)
#      Swaps = O(1)

# Average Case:
#      Comparisons = O(n²)
#      Swaps = O(n²)

# Worst Case:
#      Comparisons = O(n²)
#      Swaps = O(n²)

def bubbleSort(list_to_sort):
    for i in range(len(list_to_sort)-1,0,-1):
        for j in range(i):
            if list_to_sort[j]>list_to_sort[j+1]:
                temp = list_to_sort[j]
                list_to_sort[j] = list_to_sort[j+1]
                list_to_sort[j+1] = temp